### PR TITLE
feat: deno test --trace-ops

### PR DESCRIPTION
**Pull request by bartlomieju**
Follow up to https://github.com/denoland/deno/pull/13676

While the feature proves to be very useful it comes with a significant performance
hit, it's caused by excessive source mapping of stack frames. Given that it's non-trivial
to cache results of source mapping without unbounded consumption of memory after 
discussion with @piscisaureus we agreed that the simplest solution is
to activate this feature with a flag.

<s>Still need to add a test for it</s>

### CODE REVIEW

File: runtime/js/40_testing.js [hunk 1]
```diff
@@ -208,9 +209,16 @@
...
 
-      throw `Test case is leaking async ops.
+      let msg = `Test case is leaking async ops.
 
-- ${ArrayPrototypeJoin(details, "\n - ")}`;
+      - ${ArrayPrototypeJoin(details, "\n - ")}`;
+
+      if (!isOpCallTracingEnabled) {
+        msg +=
+          `\nTo get more details where ops are leaked run again with --trace-ops flag.`;
```
Comment (dsherret) [inline runtime/js/40_testing.js:218]: First thing I scrolled for. Good 👍

**Review by dsherret — APPROVED** [review]
LGTM and makes a lot of sense.

File: runtime/js/40_testing.js [hunk 2]
```diff
@@ -34,6 +34,7 @@
     TypeError,
   } = window.__bootstrap.primordials;
 
+  let isOpCallTracingEnabled = false;
```
Comment (dsherret) [inline runtime/js/40_testing.js:37]: Maybe we should ask the other module for this state so we have a single source of truth?

Comment (bartlomieju) [reply runtime/js/40_testing.js:37]: That was my initial thought - but there might be some setup code execute before we call `Deno[Deno.internal].runTests()` and thus if we don't do it this way then we won't be able to collect traces for the setup code.

Comment (bartlomieju) [reply runtime/js/40_testing.js:37]: Nvm, I see what you mean. I switched to using a function in `Deno.core` instead.

**Pull request merged**